FieldOpt/Hdf5SummaryReader/hdf5_summary_reader.cpp

```diff
@@ -8,27 +8,176 @@ Hdf5SummaryReader::Hdf5SummaryReader(const std::string file_path)
   DATASET_NAME_TIMES("TIMES"),
   GROUP_NAME_FLOW_TRANSPORT("FLOW_TRANSPORT"),
   DATASET_NAME_ACTIVE_CELLS("ACTIVE_CELLS"),
-  DATASET_NAME_WELL_STATES("WELL_STATES")
+  DATASET_NAME_WELL_STATES("WELL_STATES"),
+  DATASET_NAME_PRESSURE("PTZ"),
+  DATASET_NAME_SATURATION("GRIDPROPTIME")
 {
     readTimeVector(file_path);
-    readActiveCells(file_path);
     readWellStates(file_path);
+    readActiveCells(file_path);
+    readReservoirPressure(file_path);
+    readSaturation(file_path);
 }
 
-void Hdf5SummaryReader::readActiveCells(std::string file_path) {
+void Hdf5SummaryReader::readSaturation(std::string file_path) {
+
+    // Check file exists
+    H5File file(file_path, H5F_ACC_RDONLY);
+    Group group = Group(file.openGroup(GROUP_NAME_FLOW_TRANSPORT));
+    auto dataset_exists = H5Lexists(group.getId(), "GRIDPROPTIME", H5F_ACC_RDONLY);
+    // std::cout << "dataset_exists = " << dataset_exists << std::endl;
+
+    std::vector<std::vector<double>> sgas, soil, swat;
+
+    /*!
+    \todo Overall, this needs a better implementation, e.g., read soil and sgas
```

Comment:
Todo comments such as these two should be in the doxygen comment in the header file.